lib/file_transfer_agent/encrypt_util.js

```diff
@@ -125,206 +124,278 @@ function EncryptUtil(encrypt, filestream, temp) {
+      fs.readFile(inputFilePath, (err, data) => {
         if (err) {
           reject(err);
         }
-        tempOutputFileName = path;
-        tempFd = fd;
-        resolve();
+        resolve(data);
       });
     });
 
-    // Create key decipher with decoded key and AES ECB
-    let decipher = crypto.createDecipheriv(aesEcb(keySize), decodedKey, null);
-    const fileKey = Buffer.concat([
-      decipher.update(keyBytes),
-      decipher.final()
-    ]);
-
-    // Create decipher with file key, iv bytes, and AES CBC
-    decipher = crypto.createDecipheriv(aesCbc(keySize), fileKey, ivBytes);
+    const decryptedData = this.decryptGCM(fileContent, fileKey, dataIvBytes, dataAadBytes);
+    return await writeContentToFile(tmpDir, path.basename(inputFilePath) + '#', decryptedData);
+  };
+  
+  function performCrypto(cipherOrDecipher, data) {
+    const encryptedOrDecrypted = cipherOrDecipher.update(data);
+    const final = cipherOrDecipher.final();
+    return Buffer.concat([encryptedOrDecrypted, final]);
+  }
 
+  async function performFileStreamCrypto(cipherOrDecipher, tmpDir, inputFilePath, chunkSize) {
+    let outputFilePath;
+    let outputFileFd;
+    await new Promise((resolve, reject) => {
+      tmp.file({ dir: tmpDir, prefix: path.basename(inputFilePath) + '#' }, (err, path, fd) => {
+        if (err) {
+          reject(err);
+        }
+        outputFilePath = path;
+        outputFileFd = fd;
+        resolve();
+      });
+    });
     await new Promise(function (resolve) {
-      const infile = fs.createReadStream(inFileName, { highWaterMark: chunkSize });
-      const outfile = fs.createWriteStream(tempOutputFileName);
-
-      infile.on('data', function (chunk) {
-        // Dncrypt chunk using decipher
-        const decrypted = decipher.update(chunk);
-        // Write to temp file
-        outfile.write(decrypted);
+      const inputStream = fs.createReadStream(inputFilePath, { highWaterMark: chunkSize });
+      const outputStream = fs.createWriteStream(outputFilePath);
+
+      inputStream.on('data', function (chunk) {
+        const encrypted = cipherOrDecipher.update(chunk);
+        outputStream.write(encrypted);
       });
-      infile.on('close', function () {
-        outfile.write(decipher.final());
-        outfile.close(resolve);
+      inputStream.on('close', function () {
+        outputStream.write(cipherOrDecipher.final());
+        outputStream.close(resolve);
       });
     });
 
-    // Close temp file
     await new Promise((resolve, reject) => {
-      fs.close(tempFd, (err) => {
+      fs.close(outputFileFd, (err) => {
         if (err) {
-          reject(err); 
+          reject(err);
         }
         resolve();
       });
     });
+    return outputFilePath;
+  }
 
-    return tempOutputFileName;
-  };
+  async function writeContentToFile(tmpDir, prefix, content,) {
+    let outputFilePath;
+    let outputFileFd;
+    await new Promise((resolve, reject) => {
```

Comment:
let's our promises returns object/array with outputFilePath and outputFileFd in resolve instead of setting it internally

Reply:
Done